litestar/contrib/mongodb_motor/repository/types.py

```diff
@@ -0,0 +1,9 @@
+from typing import Any, MutableMapping, TypeVar
+
+from motor.motor_asyncio import AsyncIOMotorCollection
+
+__all__ = ("Document", "AsyncMotorCollection")
+
+
+Document = TypeVar("Document", bound=MutableMapping[str, Any])
```

Comment:
so, why are you binding Document to a mutablemapping? you should actually bind this to a specific type that has all the methods associated with a document. AM i missing something?

Reply:
I might be misunderstanding what you're saying here, apologies if I am, from what I understand the collection methods  such as `insert_one`, `insert_many`, take `Union[_DocumentType, RawBSONDocument]` where `_DocumentType = TypeVar("_DocumentType", bound=Mapping[str, Any])`. Should we support both `_DocumentType` and`RawBSONDocument` as inputs to the methods such as `add` and `add_many` on the repository?

Reply:
input maybe, but what about output? usually when you query a document you want to get a document instance, no?

Reply:
I've not personally used pymongo + motor before so I'm not sure what the normal conventions are. It seems most of the `find_one`, `find` type methods return `_DocumentType`, is there some other type that is normally expected to be returned?

Reply:
no, this is it